src/modules/common/components/admin-toggle.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import { connect } from 'react-redux';
+import { setAdminMode } from '../action-creators';
+
+const AdminToggle = ({ adminMode, dispatch }) => {
+  if (localStorage.getItem('adminFlag') && !adminMode) {
+    dispatch(setAdminMode(!adminMode));
+  }
+
+  const toggleAdminMode = (e) => {
+    apiClient.update({
+      'params.admin': e.target.checked || undefined,
+    });
+
+    if (e.target.checked) {
+      localStorage.setItem('adminFlag', true);
+    } else {
+      localStorage.removeItem('adminFlag');
+    }
+
+    dispatch(setAdminMode(!adminMode));
+  };
+
+  const toggleClass = adminMode ? 'footer-admin-toggle--active' : '';
+
+  return (
+    <label
+      className={`footer-admin-toggle ${toggleClass}`}
+      htmlFor="adminMode"
+    >
+      <input type="checkbox" checked={adminMode} value={adminMode} onClick={toggleAdminMode} />{' '}
+      Admin Mode
+    </label>
+  );
+};
+
+AdminToggle.propTypes = {
+  adminMode: React.PropTypes.bool,
+  dispatch: React.PropTypes.func,
+};
+
+function mapStateToProps(state) {
```

Comment:
`adminMode` should just be passed down as a prop from the parent to make this independent of redux.